MonitoringSystem/Client/MonitoringReporter.py

```diff
@@ -0,0 +1,135 @@
+             
+    self.__monitoringType = monitoringType
+  
+  def processRecords( self ):
+    """
+    It consumes all messaged from the MQ (these are failover messages). In case of failure, the messages
+    will be inserted to the MQ again.
+    """
+    try:
+      mqListener = MQListener( self.__monitoringType )
+    except MQConnectionError as exc:
+      gLogger.error( "Fail to create Listener: %s" % exc )
+      return S_ERROR( "Fail to create Listener: %s" % exc )
+     
+    result = S_OK()
+    while result['OK']:
+      # we consume all messages from the listener internal queue.
+      result = mqListener.get()
+      mqListener.stop()  # make sure that we will not proccess any more messages.
+      if result['OK']:
+        records = json.loads( result['Value'] )
+        retVal = self.__db.put( list( records ), self.__monitoringType )
+        if not retVal['OK']:
+          # the db is not available and we publish again the data to MQ
+          res = self.publishRecords( records )
+          if not res['OK']:
+            return res
+    
+    return S_OK()
+    
+  def addRecord( self, rec ):
+    """
+    It inserts the record to the list
+    :param: dict rec it kontains a key/value pair.
+    """
+    self.__documents.append( rec )
+       
+  def publishRecords( self, records ):
+    """
+    send data to the MQ
+    :param: list records contains a list of key/value pairs (dictionaries)
+    """
+    return self.__mqPublisher.put( json.dumps( records ) )
+      
+  def commit( self ):
+    """
+    It inserts the accumulated data to the db. In case of failure
+    it keeps in memory/MQ
+    """
+    self.__documentLock.acquire()
+    documents = self.__documents
+    self.__documents = []
+    self.__documentLock.release()
+    recordSent = 0
+    try:
+      while documents:
+        recordsToSend = documents[ :self.__maxRecordsInABundle ]
+        retVal = self.__db.put( recordsToSend, self.__monitoringType )
+        if retVal[ 'OK' ]:
+          recordSent += len( recordsToSend )
+          del documents[ :self.__maxRecordsInABundle ]          
+        else:
+          if self.__mq:
+            res = self.publishRecords( recordsToSend )
+            # if we managed to publish the records we can delete from the list
+            if res['OK']:
+              recordSent += len( recordsToSend )
+              del documents[ :self.__maxRecordsInABundle ]
+            else:
+              return res  # in case of MQ problem
+    except Exception as e:  # pylint: disable=broad-except
+      gLogger.exception( "Error committing", lException = e )
+      return S_ERROR( "Error committing %s" % repr( e ).replace( ',)', ')' ) )    
+    finally:
+      self.__documents.extend( documents )
+        
+    if self.__mq:
+      result = self.processRecords()
+      if not result['OK']:
+        gLogger.error( "Unable to insert data from the MQ", result['Message'] )
+        
```

Comment:
yes. We have to keep it simple as it is. The messages must be in the db level (and they are)